### ISSUE TITLE
This breaks if other attributes have delegated methods

### ISSUE THREAD

**Issue by robrwo** [tobyink/p5-moox-enumeration#1]
I thought I'd give this module a try, since it looks to be very useful. Alas, attributes with delegation seem to cause the following errors:

> Can't use string ("deleted_method_name") as an ARRAY ref while "strict refs" in use at ... local/lib/perl5/MooX/Enumeration.pm line 116.

These are hash reference delgations, e,g.

```perl
has attr => (
 ...
  handles => {
    my_name => 'delegated_method_name',
  },
)
```

or array references:

```perl
has attr => (
 ...
  handles => [qw/ delegated_method_name /],
)
```

I'm looking at the code at [line 38](https://metacpan.org/source/TOBYINK/MooX-Enumeration-0.006/lib/MooX/Enumeration.pm#L38) and wondering why it is trying to install delegates if the attribute is not actually an enum.

**Comment by tobyink** [comment]
Ooh, interesting.

I'll get this fixed ASAP.

**Comment by tobyink** [comment]
Fix included in 0.007 which should now be on CPAN.
